src/index.js

```diff
@@ -58,7 +58,7 @@ export default function loader(content) {
   if (options.publicPath !== undefined) {
     // support functions as publicPath to generate them dynamically
     publicPath = JSON.stringify(
-      typeof options.publicPath === 'function' ? options.publicPath(url) : options.publicPath + url,
+      typeof options.publicPath === 'function' ? options.publicPath(url, this._module.issuer.resource) : options.publicPath + url,
```

Comment:
Better send second params `this` and you can inside you code what you need, it is universal solution.

Reply:
The proposal is to know what is the path of the resource that called the loader. With this information any developer will get to know the level of the tree of the origin file and adapt the output path according.